src/Aspire.Hosting.Qdrant/QdrantBuilderExtensions.cs

```diff
@@ -0,0 +1,106 @@
+    /// <param name="isReadOnly">A flag that indicates if this is a read-only volume.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<QdrantServerResource> WithDataVolume(this IResourceBuilder<QdrantServerResource> builder, string? name = null, bool isReadOnly = false)
+        => builder.WithVolume(name ?? VolumeNameGenerator.CreateVolumeName(builder, "data"), "/var/lib/qdrant/storage", isReadOnly);
+
+    /// <summary>
+    /// Adds a bind mount for the data folder to a Qdrant container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="source">The source directory on the host to mount into the container.</param>
+    /// <param name="isReadOnly">A flag that indicates if this is a read-only mount.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<QdrantServerResource> WithDataBindMount(this IResourceBuilder<QdrantServerResource> builder, string source, bool isReadOnly = false)
+        => builder.WithBindMount(source, "/var/lib/qdrant/storage", isReadOnly);
+
+    /// <summary>
+    /// Adds a bind mount for the init folder to a Qdrant container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="source">The source directory on the host to mount into the container.</param>
+    /// <param name="isReadOnly">A flag that indicates if this is a read-only mount.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<QdrantServerResource> WithInitBindMount(this IResourceBuilder<QdrantServerResource> builder, string source, bool isReadOnly = true)
+        => builder.WithBindMount(source, "/docker-entrypoint-initdb.d", isReadOnly);
+
+    /// <summary>
+    /// Add a reference to a Qdrant settings for a project.
+    /// </summary>
+    /// <param name="builder">An <see cref="IResourceBuilder{T}"/> for <see cref="ProjectResource"/></param>
+    /// <param name="qdrantResource">The Qdrant server resource</param>
+    /// <returns></returns>
+    public static IResourceBuilder<ProjectResource> WithReference(this IResourceBuilder<ProjectResource> builder, IResourceBuilder<QdrantServerResource> qdrantResource)
+    {
+        builder.WithEnvironment(context =>
+        {
+            context.EnvironmentVariables[$"ConnectionStrings__{qdrantResource.Resource.Name}"] = qdrantResource.Resource.ConnectionStringExpression;
+            context.EnvironmentVariables[$"Parameters__{qdrantResource.Resource.ApiKeyParameter.Name}"] = qdrantResource.Resource.ApiKeyParameter.Value;
+
+            var restEndpointUrl = qdrantResource.Resource.GetEndpoint("rest")?.Url;
+            if (restEndpointUrl != null)
+            {
+                context.EnvironmentVariables[$"ConnectionStrings__{qdrantResource.Resource.Name}_rest"] = restEndpointUrl;
+            }
```

Comment:
We're missing an API here, ideally this would be:

```suggestion
            var restEndpointUrl = qdrantResource.Resource.GetEndpoint("rest");
            if (restEndpointUrl.Exists)
            {
                context.EnvironmentVariables[$"ConnectionStrings__{qdrantResource.Resource.Name}_rest"] = restEndpointUrl;
            }
```